### PR TITLE
38 profile page editing

### PR DESCRIPTION
**Pull request by andrewzpu**
Added edit profile page which allows users to edit and update preferred name, major, degree, graduation year, and car.
Includes new frontend page and backend post route to update database

### CODE REVIEW

**Review by ryanyychen — COMMENTED** [review]
When opening the profile edit page when 'Other' for Major or Degree is selected, the default selected should be 'Other' and what they have there should be entered.

File: myapp/api/forms.py
Comment (ryanyychen) [inline myapp/api/forms.py:1]: Please remove the comments on this page.

Comment (andrewzpu) [reply myapp/api/forms.py:1]: Do you want me to remove the lines or just uncomment them?

File: frontend/src/Pages/ProfileEdit.svelte [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+
+        // sort form data
+        // customUser
+        //    preferred_name, grad_year
+        // inductee/member
+        //    major, degree
+        // outreach
+        //    car
+
+        // maybe just give the whole form and have the view process it
```
Comment (ryanyychen) [inline frontend/src/Pages/ProfileEdit.svelte:85]: Remove comment lines 85-87. Keep comments line 77-83 tho, thanks.

**Review by ryanyychen — CHANGES_REQUESTED** [review]
Please add:

Attempts to navigate away from the page should prompt a confirmation message if anything has been changed
Add in cancel button that returns user to profile page

**Review by ryanyychen — CHANGES_REQUESTED** [review]
Great work on implementing the cancel button! Thanks Andrew.

One more thing to add: could you have the leave page confirmation pop up not just when they click on the cancel button but also any attempts to leave or refresh the page?

**Review by ryanyychen — CHANGES_REQUESTED** [review]
Thanks for pushing the change quickly. The save and cancel buttons do not seem to be clickable anymore...

**Review by ryanyychen — APPROVED** [review]
Great work, thanks!

**Pull request merged**